Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -225,9 +230,43 @@ public void run(@NotNull ProgressIndicator indicator) {
     @Override
     public void beforeDocumentChange(DocumentEvent e) {
       myServerData.onDocumentChanged(e);
+      synchronized (myLock) {
+        final Document document = e.getDocument();
+        myUpdatedDocuments.add(document);
```

Comment:
Not everyone in the world edits only Dart files.
Even if Dart  - file may be outside of project root.
Even if in project, it may be not of interest if we do not keep any regions for it.
There are also other ifs.
Checking file presence in DartServerData maps seems to solve all of them.